javascript/ql/src/experimental/Security/CWE-347-noVerification/jwtDecode.ql

```diff
@@ -0,0 +1,31 @@
+/**
+ * @name JWT missing secret or public key verification
+ * @description The application does not verify the JWT payload with a cryptographic secret or public key.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 8.0
+ * @precision high
+ * @id js/jwt-missing-verification-jwt-decode
+ * @tags security
+ *       external/cwe/cwe-347
+ */
+
+import javascript
+import DataFlow::PathGraph
+
+DataFlow::Node unverifiedDecode() {
+  result = API::moduleImport("jwt-decode").getParameter(0).asSink()
```

Comment:
You're flagging all instances where `jwt-decode` is called, independently of whether any verification happens.  
But your query-message specifically says that there is no verification?  

Also, shouldn't this just be another source in your `JsonWebToken.ql` query?